src/main/scala/logphy/DataWidthCoupler.scala

```diff
@@ -0,0 +1,108 @@
+package edu.berkeley.cs.ucie.digital
+package logphy
+
+import chisel3._
+import chisel3.util._
+
+case class DataWidthCouplerParams(
+    val inWidth: Int = 4,
+    val outWidth: Int = 4,
+)
+
+class DataWidthCouplerIO(
+    params: DataWidthCouplerParams,
+) extends Bundle {
+  val in = Flipped(Decoupled(UInt(params.inWidth.W)))
+  val out = Decoupled(UInt(params.outWidth.W))
+}
+
+class DataWidthCoupler(params: DataWidthCouplerParams) extends Module {
```

Comment:
Can you add some comment on top of this class on what this does and where it is used?